CliFx/Domain/IConverter.cs

```diff
@@ -0,0 +1,13 @@
+﻿namespace CliFx.Domain
+{
+    /// <summary>
+    /// Used as an interface for implementing custom parameter/option converters.
+    /// </summary>
+    public interface IConverter
```

Comment:
Makes sense to move it to root namespace (CliFx) because CliFx.Domain is internal. Maybe also rename it to `IArgumentValueConverter` to make it more specific to our use case?

Reply:
agree, done